benchmark/framework/parser/file_access.go

```diff
@@ -0,0 +1,153 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package bparser
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+var (
+	outputDir      = "./output/"
+	sociLogs       = outputDir + "soci-snapshotter-stderr"
+	FileAccessDir  = outputDir + "file_access_logs/"
+	containerdLogs = outputDir + "containerd-stderr"
+)
+
+type FileAccessPatterns struct {
+	ContainerStartTime time.Time   `json:"containerStartTime"`
+	Operations         []Operation `json:"operations"`
+}
+
+type BaseOperation struct {
+	Level     string    `json:"level"`
+	Msg       string    `json:"msg"`
+	Operation string    `json:"operation"`
+	Path      string    `json:"path"`
+	Time      time.Time `json:"time"`
+}
+
+type Operation struct {
+	Operation                 string `json:"operation"`
+	Path                      string `json:"path"`
+	FirstAccessTimeAfterStart string `json:"firstAccessTimeAfterStart"`
+	Count                     int    `json:"count"`
+}
+
+func ParseFileAccessPatterns(imageName string) error {
+	startTime, err := getTaskStartTime()
+	if err != nil {
+		return err
+	}
+	fa := FileAccessPatterns{
+		ContainerStartTime: *startTime,
+	}
+	file, err := os.Open(sociLogs)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err = file.Close()
+	}()
+
+	b, err := io.ReadAll(file)
+	if err != nil {
+		return err
+	}
+
+	logs := strings.Split(string(b), "\n")
+
+	m := make(map[string]Operation)
+
+	for _, log := range logs {
+		if strings.Contains(log, "FUSE") {
```

Comment:
Does it work if the file path itself is containing the word `FUSE`?

Reply:
Ahhh... This would fail in that case. I could potentially look for "FUSE operation" perhaps. Or rather unmarshal to a struct since I believe every log contains at least `msg:` field and then filter by `msg` value.

Reply:
I have also added the ability to log the total count for each fuse operation. This may be useful for determining  possible correlations between any specific fuse operation and the total benchmark time.